code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -57,47 +59,64 @@
+			forceMove(get_turf(summoner))
+			PoolOrNew(/obj/effect/overlay/temp/guardian/phase, get_turf(src))
+			//loc = get_turf(summoner)
 
 /mob/living/simple_animal/hostile/guardian/Move() //Returns to summoner if they move out of range
-	..()
+	. = ..()
+	snapback()
+			
+/mob/living/simple_animal/hostile/guardian/proc/snapback()
 	if(summoner)
-		if (get_dist(get_turf(summoner),get_turf(src)) <= range)
+		if(get_dist(get_turf(summoner),get_turf(src)) <= range)
 			return
 		else
-			src << "You moved out of range, and were pulled back! You can only move [range] meters from [summoner.real_name]"
-			visible_message("<span class='danger'>The [src] jumps back to its user.</span>")
-			loc = get_turf(summoner)
+			src << "<span class='holoparasite'>You moved out of range, and were pulled back! You can only move [range] meters from [summoner.real_name]!</span>"
+			visible_message("<span class='danger'>\The [src] jumps back to its user.</span>")
+			PoolOrNew(/obj/effect/overlay/temp/guardian/phase/out, get_turf(src))
+			forceMove(get_turf(summoner))
+			PoolOrNew(/obj/effect/overlay/temp/guardian/phase, get_turf(src))			
 
-/mob/living/mob/living/simple_animal/hostile/guardian/canSuicide()
+/mob/living/simple_animal/hostile/guardian/canSuicide()
 	return 0
 
 /mob/living/simple_animal/hostile/guardian/death()
 	..()
 	summoner << "<span class='danger'><B>Your [name] died somehow!</span></B>"
 	summoner.death()
+	
+/mob/living/simple_animal/hostile/guardian/proc/updatehudhealth()
+	if(summoner)
+		var/resulthealth
+		if(iscarbon(summoner))
+			resulthealth = round((abs(config.health_threshold_dead - summoner.health) / abs(config.health_threshold_dead - summoner.maxHealth)) * 100)
+		else
+			resulthealth = round((summoner.health / summoner.maxHealth) * 100)
+		hud_used.guardianhealthdisplay.maptext = "<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='#efeeef'>[resulthealth]%</font></div>"
 
-/mob/living/simple_animal/hostile/guardian/adjustBruteLoss(amount) //The spirit is invincible, but passes on damage to the summoner
-	var/damage = amount * damage_transfer
-	if (summoner)
+
+/mob/living/simple_animal/hostile/guardian/adjustHealth(amount) //The spirit is invincible, but passes on damage to the summoner
+	if(summoner)
 		if(loc == summoner)
 			return
-		summoner.adjustBruteLoss(damage)
-		if(damage)
+		summoner.adjustBruteLoss(amount)
+		if(amount)
 			summoner << "<span class='danger'><B>Your [name] is under attack! You take damage!</span></B>"
 			summoner.visible_message("<span class='danger'><B>Blood sprays from [summoner] as [src] takes damage!</B></span>")
 		if(summoner.stat == UNCONSCIOUS)
 			summoner << "<span class='danger'><B>Your body can't take the strain of sustaining [src] in this condition, it begins to fall apart!</span></B>"
-			summoner.adjustCloneLoss(damage)
-
+			summoner.adjustCloneLoss(amount*0.5) //dying hosts take 50% bonus damage as cloneloss
```

Comment:
`amount*0.5`
> 50% bonus damage

Shouldn't that be "only 50% damage" or did you mean to double the damage?